src/Refactoring-Core/RBInlineTemporaryRefactoring.class.st

```diff
@@ -38,55 +38,70 @@ RBInlineTemporaryRefactoring class >> model: aRBSmalltalk inline: anInterval fro
 ]
 
 { #category : 'preconditions' }
-RBInlineTemporaryRefactoring >> applicabilityPreconditions [
+RBInlineTemporaryRefactoring >> applicabilityPreconditions [ 
 
+	| validator |
+	
 	^ {
-		  (RBCondition definesSelector: selector in: class).
-		  (RBCondition withBlock: [
-			   self checkSelectedInterval.
-			   true ]) }
+		(ReDefinesSelectorsCondition new definesSelectors: (Array with: selector) in: class).
+		
+		(ReBlockCondition new 
+			block: [
+				self checkSelectedInterval.
+				validator := ReIsValidInstanceVariableName new.
+				(validator check: assignmentNode variable name)
+					ifFalse: [ 
+						self refactoringError: (validator violationMessageOn: assignmentNode variable name)].
+				true];
+			violatorErrorString: 'Check selected interval failed').		
+	}
 ]
 
 { #category : 'preconditions' }
 RBInlineTemporaryRefactoring >> checkSelectedInterval [
 
-	sourceTree := class parseTreeForSelector: selector.
-	sourceTree ifNil: [ self refactoringError: 'Could not parse source' ].
-	assignmentNode := sourceTree whichNodeIsContainedBy: sourceInterval.
-	assignmentNode isAssignment
-		ifFalse: [ self refactoringError: 'The selected node is not an assignment statement' ].
-	definingNode := assignmentNode whoDefines:
-		                assignmentNode variable name.
-	self hasOnlyOneAssignment
+	| searcher matchesCount |
+	searcher := self parseTreeSearcher.
+	matchesCount := 0.
+	searcher 
+		matches: assignmentNode variable name , ' := ``@object'
+		do: [ :aNode | matchesCount := matchesCount + 1 ].
+	(matchesCount = 1)
```

Comment:
I really like this inline of `hasOnlyOneAssignment`. Can we now go one step further and encapsulate this in a condition and write a few nice tests for it? (also, there might already be existing or similar condition for inspiration)